
syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service Category{
    // 商品分类接口
    rpc GetAllCategorysList(google.protobuf.Empty) returns (CategoryListResponse);
    rpc GetSubCategory(CategoryListRequest) returns (SubCategoryListResponse);
    rpc CreateCategory(CategoryInfoRequest) returns (CategoryInfoResponse);
    rpc DeleteCategory(DeleteCategoryRequest) returns (google.protobuf.Empty);
    rpc UpdateCategory(CategoryInfoRequest) returns (google.protobuf.Empty);
}

message CategoryListRequest {
    int32 id = 1;
    int32 level = 2;
}

message CategoryInfoRequest {
    int32 id = 1;
    string name = 2;
    int32 parentCategory = 3;
    int32 level = 4;
    bool isTab = 5;
}

message DeleteCategoryRequest {
    int32 id = 1;
}

message CategoryInfoResponse {
    int32 id = 1;
    string name = 2;
    int32 parentCategory = 3;
    int32 level = 4;
    bool isTab = 5;
}

message CategoryListResponse {
    int32 total = 1;
    repeated CategoryInfoResponse data =  2;
    string jsonData = 3;
}

message SubCategoryListResponse {
    int32 total = 1;
    CategoryInfoResponse info =2;
    repeated CategoryInfoResponse subCategorys = 3;
}
